controllers/space/space_controller.go

```diff
@@ -0,0 +1,343 @@
+	if space.Spec.TargetCluster == "" {
+		return false, r.setStatusProvisioningFailed(logger, space, fmt.Errorf("unspecified target member cluster"))
+	}
+	memberCluster, found := r.MemberClusters[space.Spec.TargetCluster]
+	if !found {
+		return false, r.setStatusProvisioningFailed(logger, space, fmt.Errorf("unknown target member cluster '%s'", space.Spec.TargetCluster))
+	}
+	name := space.Name // NSTemplateSet will have the same name as the Space
+	logger = logger.WithValues("target_member_cluster", space.Spec.TargetCluster, "name", name)
+	// create if not found
+	nsTmplSet := &toolchainv1alpha1.NSTemplateSet{}
+	if err := memberCluster.Client.Get(context.TODO(), types.NamespacedName{
+		Namespace: memberCluster.OperatorNamespace,
+		Name:      space.Name,
+	}, nsTmplSet); err != nil {
+		if errors.IsNotFound(err) {
+			logger.Info("creating NSTemplateSet on target member cluster")
+			if err := r.setStatusProvisioning(space); err != nil {
+				return false, err
+			}
+			if nsTmplSet, err = r.newNSTemplateSet(memberCluster.OperatorNamespace, space); err != nil {
+				return false, r.setStatusProvisioningFailed(logger, space, err)
+			}
+
+			if err := memberCluster.Client.Create(context.TODO(), nsTmplSet); err != nil {
+				logger.Error(err, "failed to create NSTemplateSet on target member cluster")
+				return false, r.setStatusNSTemplateSetCreationFailed(logger, space, err)
+			}
+			logger.Info("NSTemplateSet created on target member cluster")
+			return false, nil
+		}
+		return false, r.setStatusNSTemplateSetCreationFailed(logger, space, err)
+	}
+	logger.Info("NSTemplateSet already exists")
+
+	readyCond, found := condition.FindConditionByType(nsTmplSet.Status.Conditions, toolchainv1alpha1.ConditionReady)
+	if !found || readyCond.Status != corev1.ConditionTrue {
+		logger.Info("NSTemplateSet is not ready", "ready-condition", readyCond)
+		return true, nil // here we need to explicitly requeue since the controller doesn't watch the NSTemplateSetStatus
+	}
+	return false, r.setStatusReady(space)
+}
+
+func (r *Reconciler) newNSTemplateSet(memberOperatorNS string, space *toolchainv1alpha1.Space) (*toolchainv1alpha1.NSTemplateSet, error) {
+	// look-up the NSTemplateTier
+	tmplTier := &toolchainv1alpha1.NSTemplateTier{}
+	if err := r.Client.Get(context.TODO(), types.NamespacedName{
+		Namespace: space.Namespace,
+		Name:      space.Spec.TierName,
+	}, tmplTier); err != nil {
+		return nil, err
+	}
+	// create the NSTemplateSet from the NSTemplateTier
+
+	nsTmplSet := &toolchainv1alpha1.NSTemplateSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: memberOperatorNS,
+			Name:      space.Name,
+			Finalizers: []string{
+				toolchainv1alpha1.FinalizerName,
+			},
+		},
+		Spec: *usersignup.NewNSTemplateSetSpec(tmplTier),
+	}
+	for i, tmpl := range tmplTier.Spec.Namespaces {
+		nsTmplSet.Spec.Namespaces[i] = toolchainv1alpha1.NSTemplateSetNamespace{
+			TemplateRef: tmpl.TemplateRef,
+		}
+	}
```

Comment:
This is not needed, correct? everything should be set by `usersignup.NewNSTemplateSetSpec`
```suggestion
```

Reply:
already removed in https://github.com/codeready-toolchain/host-operator/pull/548/commits/d44ab408cb72a5dd0375480e5e40aed42a4ece1d